src/proto/streams/streams.rs

```diff
@@ -352,6 +362,15 @@ impl<B> StreamRef<B>
         })
     }
 
+    pub fn is_recv_empty(&self) -> bool {
+        let mut me = self.inner.lock().unwrap();
+        let me = &mut *me;
+
+        let stream = me.store.resolve(self.key);
+
+        stream.state.is_recv_closed() && stream.pending_recv.is_empty()
```

Comment:
`stream.pending_recv` could only contain a trailers frame, and `is_recv_empty()` would return false, yet there are no more data frames?